src/main/java/pl/mimuw/zpp/quantumai/backendui/controller/PackageController.java

```diff
@@ -0,0 +1,49 @@
+package pl.mimuw.zpp.quantumai.backendui.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import pl.mimuw.zpp.quantumai.backendui.model.GraphPackage;
+import pl.mimuw.zpp.quantumai.backendui.service.PackageService;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/package")
+@RequiredArgsConstructor
+public class PackageController {
+    private final PackageService packageService;
+
+    @PostMapping("/new")
+    public ResponseEntity<String> createNewPackage(
+            @RequestParam String name
+    ) {
+        return ResponseEntity.ok(
+                packageService.createEmptyPackage(name)
+        );
+    }
+
+    @GetMapping("/all")
+    public ResponseEntity<List<GraphPackage>> getPackages() {
+        return ResponseEntity.ok(
+                packageService.getPackages()
+        );
+    }
+
+    @GetMapping("/")
+    public ResponseEntity<GraphPackage> getPackage(
+            @RequestParam String packageId
+    ) {
+        return ResponseEntity.ok(
+                packageService.getPackage(packageId)
+        );
+    }
+
+    @PostMapping("/add")
+    public ResponseEntity<Void> getPackage(
+            @RequestParam String packageId,
+            @RequestBody List<String> graphIds) {
+        packageService.addGraphsToPackage(packageId, graphIds);
+        return ResponseEntity.ok(null);
+    }
+}
```

Comment:
Wolałbym dodawanie paczki jako podanie pełnego jsona np. już razem z grafami. Dodatkowo dodawanie grafów też można zrealizować w ten sposób, więc będę mógł również usuwać grafy z paczki.

Reply:
@zemdam w jaki sposób widzisz usuwanie grafów z paczki?

Reply:
Chcąc zmienić jakąś paczkę, wysyłam po prostu nową listę grafów dla tej paczki i nią nadpisuję poprzednią.

Reply:
Źle sprecyzowałem 'listę grafów' chodziło mi o listę z id grafów. Dodałem już commita zmieniającego tworzenie paczek. A jeśli chcemy mieć edycję paczki to myślę o tym w taki sposób, że podajemy ID paczki w url lub w json oraz nową listę id grafów w json (opcjonalnie jeszcze zmiana nazwy). Tylko jeśli chcemy mieć taką funkcjonalność to musimy się zastanowić co ma się dziać z już przetworzonymi/przetwarzanymi testami.